buttplug4j/src/main/java/io/github/blackspherefollower/buttplug4j/client/ButtplugClientDevice.java

```diff
@@ -382,6 +384,116 @@ public final Future<ButtplugMessage> sendLinearCmd(final Pair<Double, Long>[] ve
                 values.toArray(new LinearCmd.LinearSubCmd[]{}), client.getNextMsgId()));
     }
 
+    /**
+     * Sends a command to read a sensor value from the device.
+     * <p>
+     * This method constructs and sends a sensor read command to the device for the specified sensor type and index.
+     * It checks if the device supports the "SensorReadCmd" attribute before attempting to send the command. 
+     * If the device does not support this attribute, or if the command cannot be sent, a {@link ButtplugDeviceException} 
+     * is thrown.
+     * </p>
+     * 
+     * @param sensorIndex The index of the sensor feature to read. This value specifies which sensor data 
+     *              to retrieve from the device.
+     * @param sensorType The type of sensor to read (e.g., "Battery"). This value indicates the specific 
+     *                   type of sensor data requested.
+     * 
+     * @return A {@link Future} representing the pending result of the sensor read command.
+     *         Once completed, the {@link ButtplugMessage} returned by the Future will contain 
+     *         the sensor data if the command succeeds.
+     * 
+     * @throws ButtplugDeviceException if the device does not support "SensorReadCmd" or if 
+     *                                 the sensor read command could not be created or sent.
+     */
+    public final Future<ButtplugMessage> sendSensorReadCmd(final int sensorIndex, final String sensorType)
+            throws ButtplugDeviceException {
+
+        MessageAttributes attrs = getDeviceMessages().get("SensorReadCmd");
+        if (!(attrs instanceof SensorMessageAttributes)) {
+            throw new ButtplugDeviceException("Device doesn't support SensorReadCmd!");
+        }
+
+        final SensorReadCmd cmd = new SensorReadCmd(this.deviceIndex, ButtplugConsts.DEFAULT_MSG_ID);
```

Comment:
Ah, just realised the messageId has to be set from `client.getNextMsgId()`: the response will have the same messageId, which is what ties up the future handling.